### PR TITLE
Exams Karolina Kosenko(IM-23)

### PR DESCRIPTION
**Pull request by karoll1naa**

### CODE REVIEW

**Comment by stasyukhymenko** [comment]
I have no idea what to change. Great job

**Comment by karoll1naa** [comment]
Це перезалив завдань, так як минулий пул реквест закрився

**Comment by tshemsedinov** [comment]
where are the functions?

File: exams/password.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const int string_length = sizeof(alphabet) ;
+int main()
+{
+    srand(time(NULL));
+    int n;
+    cout<<"Enter the length of your password:";
+    cin>>n;
+    cout<<"Generated random password:";
+    for(int i = 0; i < n; i++)
+        cout << alphabet[rand() % string_length];
```
Comment (tshemsedinov) [inline exams/password.cpp:14]: You have no function but you have no even a variable where password will be placed after loop execution

File: exams/sum.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    cout<<"Enter size Array = "<<endl;
+    cin>>i;
+    cout<<"Array: "<<endl;
+    for (int j = 0; j < i; j++){
+        cin>>A[i];
+        Sum+= A[i];
+    }
+    cout<<"Sum: "<<endl;
+    cout << Sum <<endl;
+
```
Comment (tshemsedinov) [inline exams/sum.cpp:15]: That's carpet of code :smiling_face_with_tear:

File: exams/password.cpp (hunk 1 above)
Comment (karoll1naa) [reply exams/password.cpp:14]: > You have no function but you have no even a variable where password will be placed after loop execution

Ось перероблений код
#include <iostream>
#include<ctime>
using namespace std;
const char alphabet[] = "0123456789!@#$%^&*()qwertyuiop[]asdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM";
const int string_length = sizeof(alphabet) ;
int main()
{
    srand(time(NULL));
    int n;
    cout<<"Enter the length of your password:";
    cin>>n;
    cout<<"Generated random password:";
    char *password = (char *)malloc(n+1);
    for(int i = 0; i < n; i++)
        password[i] = alphabet[rand() % string_length];
    password[n] = 0;
    cout << password;
}

Comment (tshemsedinov) [reply exams/password.cpp:14]: We need a function `generatePassword` returning array of char or even `std::string`

Comment (karoll1naa) [reply exams/password.cpp:14]: > We need a function `generatePassword` returning array of char or even `std::string`

перероблений код №2
#include <iostream>
#include<ctime>
using namespace std;

char* generatePassword(int n) {
    const char alphabet[] = "0123456789!@#$%^&*()qwertyuiop[]asdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM";
    const int string_length = sizeof(alphabet) ;
    char *password =  (char *)malloc(n+1);
    for(int i = 0; i < n; i++)
        password[i] = alphabet[rand() % string_length];
    password[n] = 0;
    return password;
}

int main()
{
    srand(time(NULL));
    int n;
    cout<<"Enter the length of your password:";
    cin>>n;
    cout<<"Generated random password:";
    char *pass = generatePassword(n);
    if (pass)
        cout << pass;
    free(pass);
    return 0;
}

**Pull request closed without merging**